source/adapters/hip/usm.cpp

```diff
@@ -190,9 +190,28 @@ urUSMGetMemAllocInfo(ur_context_handle_t hContext, const void *pMem,
 #endif
       return ReturnValue(UR_USM_TYPE_UNKNOWN);
     }
-    case UR_USM_ALLOC_INFO_BASE_PTR:
-    case UR_USM_ALLOC_INFO_SIZE:
+    case UR_USM_ALLOC_INFO_BASE_PTR: {
+#if (HIP_VERSION_MAJOR >= 5)
+      void *Base;
```

Comment:
I tried to reformat the codes for the check error. 

Should we initialize these values in the CUDA and HIP versions ?

Reply:
I wasn't aware they are uninitialized there as well, sorry. I guess it would be helpful, build setups can fail with `Werror` and unitialized variable warnings.